### DISCUSSION TITLE
getUnobservedData() from 'this'

### DISCUSSION THREAD

**Discussion by lux** [alpinejs/alpine#618] [4. General]
Is there a way to get the unobserved data (I want to serialize the whole thing to JSON) within a function in the component? The only way I see is doing something like this, which feels like there may be a more direct way that I'm just not seeing:

```js
document.querySelectorAll('[x-data]').forEach(el => {
  console.log(el.__x.getUnobservedData());
});
```

Thanks!

**Comment by ryangjchandler** [comment]
If you're inside of a component, you have access to `$el` which is the root component element, so you could just do `$el.__x.getUnobservedData()`.

**Reply by lux** [reply]
If I define a custom function, do I have to pass $el in order to access it there? I see `this` but not `$el` available there. For ex:

```js
function myTodos () {
    todos: [],
    todoCount() {
        // Can I get $el here without adding it as a function parameter?
    }
}
```

I'm working on an integration of Alpine.js into a project and want to make a simple sync() function without requiring them to add an $el parameter to each of their custom functions if possible.

**Reply by ryangjchandler** [reply]
`$el` should be available on the `this` context, `this.$el`.

**Reply by lux** [reply]
Could have sworn I'd tried that... Thanks for the help, that's exactly what I needed!
